### DISCUSSION TITLE
It is impossible to use Modal.Stack with custom compound Modal

### DISCUSSION THREAD

**Discussion by Foprta** [mantinedev/mantine#7155] [Feedback]
`Modal.Root` do not have inner logic to hide itself if rendered inside `Modal.Stack`. And we cannot do it by ourselves because `useModalStackContext` is not exported properly from the library. So usage of `Modal.Stack` is completely useless while having custom modals.

```
import { useModalStackContext } from "node_modules/@mantine/core/lib/components/Modal/ModalStack";
```

**Comment by rtivital** [comment]
Yes, currently it is not possible, I'll see if there is an option to add this feature and provide an update with the next patch release.

**Reply by Foprta** [reply]
Hi! Any updates?

**Reply by rtivital** [reply]
I've checked it and, unfortunately, it is not possible to implement this feature in compound Modal components. It was added to the documentation.

**Reply by Foprta** [reply]
Okay, but is it possible to export `useModalStackContext` correctly, so we can build it on our own?

**Reply by rtivital** [reply]
Sure, I can export that in next patch release (`7.15.1`)
